src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+   */
+  backgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Background color to show when button is being hovered over and is not disabled
+   */
+  hoverBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * The content to show inside the button
+   *
+   * When using an icon, use the `icon` prop instead.
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Class name to passthrough to the top level component
+   */
+  className?: string;
+
+  /**
+   * Disable the button
+   *
+   * This will prevent all clicks and will show the button in a different color
+   */
+  disabled?: boolean;
+
+  /**
+   * Background color to display when the button is disabled
+   */
+  disabledBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Text color to display when the button is disabled
+   */
+  disabledTextColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Treat this button as a floating action button
+   *
+   * You must include an `icon` prop and you must _not_ include a `children`
+   * prop for a floating action button.
+   */
+  fab?: boolean;
+
+  /**
+   * Force the button to be in the active state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceActiveState?: boolean;
+
+  /**
+   * Force the button to be in the focus state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceFocusState?: boolean;
+
+  /**
+   * Force the button to be in the hover state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceHoverState?: boolean;
+
+  /**
+   * A hidden button will have no background
+   *
+   * You should configure the `hiddenActiveBackgroundColor` prop if you are
+   * using any custom colors.
+   */
+  hidden?: boolean;
+
+  /**
+   * Color to make the background for a hidden button that is active.
+   *
+   * This color should be darker than the `hoverBackgroundColor`.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenActiveBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Color to make the text for a focused hidden button.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenFocusedColor?: CSS.ColorProperty;
+
+  /**
+   * Either an icon to show to the left of the button text, or on it's own
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Size of the button
+   *
+   * Defaults to "default"
+   */
+  size?: "default" | "small" | "large";
+
+  /**
+   * CSS styles to apply
+   *
+   * @ignore
+   */
+  style?: React.CSSProperties;
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  backgroundColor = colors.silver.light,
+  children,
+  disabled = false,
+  disabledBackgroundColor = colors.silver.dark,
+  disabledTextColor = colors.grey.light,
```

Comment:
we should probably decide on whether we use styled system, or at least themeprovider (imo we should), and if we do then these would make sense in the theme context instead of ad-hoc props